### PR TITLE
Strip invisible delimiters from type name

### LINKED ISSUE #155: Strip pretty-printed invisible delimiters for type names
See https://github.com/rust-lang/rust/issues/97076 and https://github.com/rust-lang/rust/pull/96682.

This can happen where `TypeInfo` is derived for a type generated by a `ty` expression in a declarative macro. This is parsed as a `syn::Type::Group`, which is then pretty-printed with the invisible delimiters `"/*»*/"` and `"/*«*/"`.

These should be stripped, possibly in `clean_type_string`, but maybe there is a cleaner way to do it.

### PR DESCRIPTION
**Pull request by ascjones**
Closes #155

### CODE REVIEW

**Review by bkchr — APPROVED** [review]
Nice and even with a test :heart:

File: test_suite/tests/derive.rs [hunk 1]
```diff
@@ -867,3 +867,24 @@ fn ranges() {
 
     assert_type!(Rangey, ty);
 }
+
+#[test]
+fn strips_invisible_delimiters_from_type_name() {
+    // When a type is specified as part of a macro, the tokens passed to the `TypeInfo` derive macro
+    // are a type `Group`, which is pretty printed with invisible delimiters e.g. /*«*/ bool /*»*/.
```
Comment (dvdplm) [inline test_suite/tests/derive.rs:874]: Maybe move/copy this comment into the code as well? Seems like a non-obvious gotcha?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.1.2] - 2022-05-18
+
+### Fixed
+- Strip invisible delimiters from type name [(#156)](https://github.com/paritytech/scale-info/pull/156)
```
Comment (dvdplm) [inline CHANGELOG.md:13]: Should be under `unreleased` maybe?

**Review by dvdplm — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (ascjones) [reply CHANGELOG.md:13]: Yeah but I was planning on releasing straight from this commit to avoid a separate PR

**Review by lexnv — APPROVED** [review]

**Comment by ascjones** [comment]
https://crates.io/crates/scale-info/2.1.2

**Pull request merged**